hnn_core/tests/test_network.py

```diff
@@ -262,6 +262,19 @@ def test_network_drives():
         n_drive_cells = net.external_drives[dn]['n_drive_cells']
         assert len(net.gid_ranges[dn]) == n_drive_cells
 
+    # Source gids should be assigned in order according to net.cell_types
+    # For a cell-specific drive, connections are made between disjoint sets of
+    # source gids by target type
+    drive_src_list = list()
+    for target_type in net.cell_types:
+        conn_idxs = pick_connection(net, src_gids='evprox1',
+                                    target_gids=target_type)
+        src_set = set()
+        for conn_idx in conn_idxs:
+            src_set.update(net.connectivity[conn_idx]['src_gids'])
+        drive_src_list.extend(sorted(src_set))
+    assert np.array_equal(drive_src_list, sorted(drive_src_list))
```

Comment:
glad you god this passing! But we should definitely bookmark checking why this doesn't work when instantiated just with `Network`